added Enabled Notification Toggle. When disabled, Sound and Vibration…

t-lam opened the pull request:
… toggles are disabled. There seems to be a lag with our emulator but the states of the toggles are saved correctly. The previous pull request that was merged was incorrect. @rhatchet

t-lam committed:
added Enabled Notification Toggle. When disabled, Sound and Vibration toggles are disabled

t-lam committed:
changed item lists to item-group for better grouping of objects in settings

kevbui commented on src/pages/page4/page4.html:
```diff
@@ -39,24 +30,20 @@
-        <ion-option value="ring">Ring</ion-option>
-        <ion-option value="alarm">Alarm</ion-option>
-      </ion-select>
+      <ion-toggle [(ngModel)] = "sound_toggle" name="sound_toggle" value="sound_toggle" (ionChange)="toggleSound()" [disabled]="!notification_toggle"></ion-toggle>
```
The toggles don't seem to lag me in the emulator, and it runs fine on my phone. So it should be ok to leave them enabled.

kevbui commented on src/pages/page4/page4.html:
```diff
@@ -19,18 +18,10 @@
-
-    <ion-list-header>Notifications</ion-list-header>
+  <ion-item-group>
+    <ion-item-divider color="light">Notifications</ion-item-divider>
```
The divider looks really weird here. Maybe it should be before the Notifications text.

kevbui commented on src/pages/page4/page4.ts:
```diff
@@ -35,65 +33,25 @@ export class Page4 {
     });
 
+    this.storage.get('sound_toggle').then((value) => {
+      this.sound_toggle = value;
```
So instead of grabbing all these variables from memory, you create one variable (object) to hold the values and just grab that. I also recommend that you use the user provider which handles all of this.

t-lam committed:
changed groups back to lists...Send Feedback button now prompts for feedback message. still need to work on send data to an email address

t-lam committed:
added enableSound to userSettings in user.ts. updated page4.ts to use user provider

t-lam committed:
Send feedback button now opens a prompt window to send a message. currently it opens up the email client on the user's phone and populates the data and fields and the user must send through the email client. we'll work on getting this to either do it in the background or to be filled on a form somewhere instead. user.ts was modified to set userSettings to defaultSettings because there is a bug with grabbing null objects from storage

t-lam committed:
fixed logout button and send feedback button

t-lam committed:
added enableSound in defaultSettings, minor text changes to logout alert

sweeney6 committed:
page4.ts:  Added ionViewWillEnter() function to handle saving the notification/sound/vibration
user.ts:  deleted the {} when setting the usersettings variable

kevbui committed:
Style fixes
